speaker/layer2_controller.go

```diff
@@ -122,11 +116,13 @@ func (c *layer2Controller) ShouldAnnounce(l log.Logger, name string, toAnnounce
 	})
 
 	// Are we first in the list? If so, we win and should announce.
-	if len(availableNodes) > 0 && availableNodes[0] == c.myNode {
+	owner := availableNodes[0] // we have returned if len()==0
+	if owner == c.myNode {
+		level.Debug(l).Log("event", "shouldannounce", "protocol", "l2", "node", c.myNode, "service", name)
 		return ""
 	}
 
-	// Either not eligible, or lost the election entirely.
+	level.Debug(l).Log("event", "skipping should announce l2", "node", c.myNode, "service", name, "reason", "not owner", "owner", owner)
```

Comment:
I have doubts about these. I think they are already covered by the "serviceWithdrawn" / Service announcing logs outside. The only extra value provided is the owner name, which _maybe_ we can append to the return value. I'd say let's remove for now.

Reply:
removed